src/strands_tools/a2a_client.py

```diff
@@ -0,0 +1,248 @@
+        if url in self._discovered_agents:
+            return self._discovered_agents[url]
+
+        resolver = await self._create_a2a_card_resolver(url)
+        agent_card = await resolver.get_agent_card()
+        self._discovered_agents[url] = agent_card
+        logger.info(f"Successfully discovered and cached agent card for {url}")
+
+        return agent_card
+
+    @tool
+    def discover_agent(self, url: str) -> dict[str, Any]:
+        """
+        Discover an A2A agent and return its agent card with capabilities.
+
+        This function fetches the agent card from the specified A2A agent URL
+        and caches it for future use.
+
+        Args:
+            url: The base URL of the A2A agent to discover
+
+        Returns:
+            dict: Discovery result including:
+                - success: Whether the operation succeeded
+                - agent_card: The full agent card data (if successful)
+                - error: Error message (if failed)
+                - url: The agent URL that was queried
+        """
+        return self._run_async(self._async_discover_agent_card_tool(url))
+
+    async def _async_discover_agent_card_tool(self, url: str) -> dict[str, Any]:
+        """Internal async implementation for discover_agent_card tool."""
+        try:
+            agent_card = await self._async_discover_agent_card(url)
+            return {
+                "status": "success",
+                "agent_card": agent_card.model_dump(mode="python", exclude_none=True),
+                "url": url,
+            }
+        except Exception as e:
+            logger.exception(f"Error discovering agent card for {url}")
+            return {
+                "status": "error",
+                "error": str(e),
+                "url": url,
+            }
+
+    @tool
+    def list_discovered_agents(self) -> dict[str, Any]:
+        """
+        List all discovered A2A agents and their capabilities.
+
+        Returns:
+            dict: Information about all discovered agents including:
+                - success: Whether the operation succeeded
+                - agents: List of discovered agents with their details
+                - total_count: Total number of discovered agents
+        """
+        try:
+            agents = [
+                agent_card.model_dump(mode="python", exclude_none=True)
+                for agent_card in self._discovered_agents.values()
+            ]
+            return {
+                "status": "success",
+                "agents": agents,
+                "total_count": len(agents),
+            }
+        except Exception as e:
+            logger.exception("Error listing discovered agents")
+            return {
+                "status": "error",
+                "error": str(e),
+                "total_count": 0,
+            }
+
+    @tool
+    def send_message(self, message_text: str, target_agent_url: str, message_id: str | None = None) -> dict[str, Any]:
```

Comment:
I think the proposed mcp tool (#49) and the use_browser tool (#88) all use overloaded tools with actions instead of individual messages. 

Did we want to mimic that and/or what are the pros and cons?